### PR TITLE
feat: usual container goodies for RbTree

### PR DESCRIPTION
**Pull request by roman-kashitsyn**
This change implements iterators for RbTree, which enables a simple
implementation of useful container traits: (Partial)Eq, (Partial)Ord.

This change also adds implementations for Clone, Debug, and FromIterator traits.

### CODE REVIEW

File: src/ic-certified-map/src/rbtree.rs [hunk 1]
```diff
@@ -189,15 +190,158 @@ impl<K: 'static + AsRef<[u8]>, V: AsHashTree + 'static> Node<K, V> {
...
 }
 
-/// Implements mutable Leaf-leaning red-black trees as defined in
+enum Position {
+    Left,
+    Middle,
+    Right,
+}
+
+enum IterState<'a, K, V> {
```
Comment (ulan) [inline src/ic-certified-map/src/rbtree.rs:199]: Could you please add invariants for the state?
The last entry in `Deep`:
(p, Left) => the left subtree of p is visited
(p, Middle) => the left subtree of p and p itself are visited
(p, Right) => p and its subtrees are visited.

**Review by ulan — APPROVED** [review]
Very clean implementation, lgtm!

File: src/ic-certified-map/src/rbtree.rs (hunk 1 above)
Comment (roman-kashitsyn) [reply src/ic-certified-map/src/rbtree.rs]: Good point, documented the invariants for Iter.

File: src/ic-certified-map/src/rbtree.rs [hunk 2]
```diff
@@ -189,15 +190,158 @@ impl<K: 'static + AsRef<[u8]>, V: AsHashTree + 'static> Node<K, V> {
...
+
+pub struct Iter<'a, K, V>(IterState<'a, K, V>);
+
+impl<'a, K, V> std::iter::Iterator for Iter<'a, K, V> {
+    type Item = (&'a K, &'a V);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        fn descend_left<'a, K, V>(
+            parents: &mut Vec<(&'a Node<K, V>, Position)>,
+            mut p: &'a Node<K, V>,
```
[the remark this answers is not in the document]
Comment (roman-kashitsyn) [reply src/ic-certified-map/src/rbtree.rs]: Change the code to use longer names: `tip` and `parent`.

**Comment by ulan** [comment]
The new version lgtm!

**Pull request merged**